### PR TITLE
Amphibious mode toggle for Salem

### PR DESCRIPTION
**Pull request by Strogoo**
Requires new exe (https://github.com/FAETHER/FA-Binary-Patches/pull/5)

Some technicals details is here -> `engine/Patches.lua`.

~~In the future all balance/script changes to `urs0201` should be also applied to `urs0204`. Only real difference between them is MotionType (RULEUMT_AmphibiousFloating and RULEUMT_Water)~~
bp copying is done via assembly now

image

#1328

### CODE REVIEW

**Comment by KionX** [comment]
Is there a guarantee that the relative address of the blueprint will not change?

Better to make a separate function.
Add a selection of blueprint by name.

**Comment by Strogoo** [comment]
Well, with current bp names the relative adress will always be the same. urs0201 + 3 blueprints (and 3 blueprints = 1200h). But I actually don't like the idea with dummy bp in lua, so I'll remake it in assembly I think. A simple memcpy that copy urs0201 to our new exxt section (thanks to Kyryll) and then changing MotionType there and switching unit's bp pointer to our new dummy blueprint.

As for new function, I have no idea how to implement this and don't want to spend time on it. Sim version of unit:GetStat already does everything that we need + it's not used anywhere in lua code and we are free to do anything we want with it. Have to mention that it's still 100% functional after the patch and you can still get unit:GetStat("KILLS") etc.

**Comment by KionX** [comment]
Mismatch of name and functionality is bad practice.
I have experience adding a sim version of SetFocusArmy, it was not difficult.

File: units/URS0204/URS0204_script.lua [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+            if self:GetCurrentLayer() ~= 'Land' then
+                self:GetStat("h1_SetSalemAmph", 0)
+            else
+                self:SetScriptBit('RULEUTC_WeaponToggle', true)
+            end  
+        end     
+    end,
+}
+
+TypeClass = URS0201
```
Comment (norraxx) [inline units/URS0204/URS0204_script.lua]: Do we really need to copy all that code, or there is the way to add on existing definition of `URS0201` bit methods?

Comment (norraxx) [reply units/URS0204/URS0204_script.lua]: Shouldn't here be different number `URS0201` -> `URS0204`? Like file is named different too.

Comment (Strogoo) [reply units/URS0204/URS0204_script.lua]: yep. Anyway, I'll try to get rid of this dummy `urs0204` bp and do all stuff in assembly

**Comment by shalkya** [comment]
ok i tested works pretty well. Some side cases where you toggle off in shallow water, will lead to destro still crossing on land. Apart from this i didn't notice anything wrong.
if someone wants to review/test a bit more, he is welcomed. I will merge it in few days.

**Comment by shalkya** [comment]
tested by nojoke

**Pull request merged**